src/Xenial.Tasty.Cli/Views/MainWindow.ViewModel.cs

```diff
@@ -93,6 +102,124 @@ public async Task ExecuteCommand(CommandItem commandItem)
                 CommandName = commandItem.Command.Name
             });
         }
+        private void WriteLine(string line = "")
+        {
+            //TODO: once we have the last exitCode and isRunning props we need to report those
+            ((IProgress<(string line, bool isRunning, int exitCode)>)LogProgress).Report((line, true, 0));
+        }
+
+        public Task ReportSummary(IEnumerable<SerializableTestCase> tests)
+        {
+            var totalTests = tests.Count();
+            var failedTests = tests.Where(m => m.TestOutcome == TestOutcome.Failed).Count();
+            var ignoredTests = tests.Where(m => m.TestOutcome == TestOutcome.Ignored).Count();
+            var notRunTests = tests.Where(m => m.TestOutcome == TestOutcome.NotRun).Count();
+            var successTests = tests.Where(m => m.TestOutcome == TestOutcome.Success).Count();
+            var outcome = tests.Where(t => t.TestOutcome > TestOutcome.Ignored).Min(t => t.TestOutcome);
+
+            var totalTime = tests.Sum(m => m.Duration);
+            var failedTime = tests.Where(m => m.TestOutcome == TestOutcome.Failed).Sum(m => m.Duration);
+            var ignoredTime = tests.Where(m => m.TestOutcome == TestOutcome.Ignored).Sum(m => m.Duration);
+            var notRunTime = tests.Where(m => m.TestOutcome == TestOutcome.NotRun).Sum(m => m.Duration);
+            var successTime = tests.Where(m => m.TestOutcome == TestOutcome.Success).Sum(m => m.Duration);
+
+            var totalTimeString = totalTime.AsDuration();
```

Comment:
Perhaps you could replace all this with a single enumeration?
`            var counts = new Dictionary<TestOutcome, int>();
            var durations = new Dictionary<TestOutcome, TimeSpan>();
            var totalTime = TimeSpan.Zero;
            foreach (var test in tests)
            {
                if (counts.ContainsKey(test.TestOutcome))
                {
                    counts[test.TestOutcome]++;
                    durations[test.TestOutcome] += test.Duration;
                }
                else
                {
                    counts.Add(test.TestOutcome, 1);
                    durations.Add(test.TestOutcome, test.Duration);
                }

                totalTime += test.Duration;
            }`

Reply:
This would basically be a `GroupBy` and be same efficient without sacrificing readablity. But we should introduce some helper methods for that kind of reporting easier. We should address this in #51.